### PR TITLE
Rework Kali Template documentation

### PR DESCRIPTION
**Pull request by neowutran**
My issues with the current Kali template documentation
-------------------------------------------------------------------------
- Contains and start with obsolete QubesOS version (r3.X)
- The "Katoolin" tools haven't been updated for quite some times, never succeeded to create a Kali template with it.
- The package "kali-linux-full" doesn't exist anymore

Main modification I am trying to make
--------------------------------------------------
* Delete all the obsolete part, and keep only documentation for 4.0, and only the method "from the iso" and the method "from a debian template"
* Try to explain the advantages and disadvantages of both way of creating a Kali VM, and recommend one

### CODE REVIEW

**Comment by andrewdavidwong** [comment]
@unman, do you feel comfortable reviewing this?

**Comment by unman** [comment]
On Sun, May 10, 2020 at 06:56:55AM -0700, Andrew David Wong wrote:
> @unman, do you feel comfortable reviewing this?
> 

Yes
Tomorrow

File: external/os-guides/pentesting/kali.md
Comment (unman) [inline external/os-guides/pentesting/kali.md:48]: Neither of these changes is necessary.
PGP key is correct, not GPG key.
*Some* article is needed here.

File: external/os-guides/pentesting/kali.md
Comment (unman) [inline external/os-guides/pentesting/kali.md:15]: such a template

File: external/os-guides/pentesting/kali.md
Comment (unman) [inline external/os-guides/pentesting/kali.md:16]: Kali Linux distribution is a rolling distribution based on Debian testing release, so it will always have a newer software base than available in the Qubes OS Debian template. Keep in mind that this may result in problems (especially in regard to package dependencies) not covered by this tutorial.

File: external/os-guides/pentesting/kali.md
Comment (unman) [inline external/os-guides/pentesting/kali.md:44]: PGP key

File: external/os-guides/pentesting/kali.md
Comment (unman) [inline external/os-guides/pentesting/kali.md:62]: unnecessary - "in" is correct

File: external/os-guides/pentesting/kali.md
Comment (unman) [inline external/os-guides/pentesting/kali.md:72]: templates should be template

File: external/os-guides/pentesting/kali.md
Comment (unman) [inline external/os-guides/pentesting/kali.md:79]: What is the *latest* doing here?

File: external/os-guides/pentesting/kali.md
Comment (unman) [inline external/os-guides/pentesting/kali.md:160]: Make it clear it is system storage of the Template that needs to be resized.

File: external/os-guides/pentesting/kali.md
Comment (unman) [inline external/os-guides/pentesting/kali.md:116]: Escaping is ugly - you can use *any* delimiter in a sed statement, so s%bullseye/updates%bullseye-security% is a little clearer.

File: external/os-guides/pentesting/kali.md
Comment (neowutran) [reply external/os-guides/pentesting/kali.md]: " Some article is needed here." where exactly ? 
The other points should be OK now

and thanks ! : )

Comment (unman) [reply external/os-guides/pentesting/kali.md]: "check key you download" is ugly

**Comment by 0rb677** [comment]
There is a "Dependency hell" if you try install kali-linux-default on top of debian-10 or bullseye...

**Comment by neowutran** [comment]
@0rb677 
You are right, i will adjust the doc to reduce the 'dependency hell' to be able to install 'kali-linux-default' easily. Won't be perfect however

**Comment by 0rb677** [comment]
@neowutran sorry for oftop. Did you try to build fresh archlinux template? There is `vmm-xen` build fails. I think you are a archlinux  maintainer now. Can you see it too, please?

**Comment by neowutran** [comment]
@0rb677 if you can check the changes, the 'dependency hell' should be more acceptable now. 
@unman the commit 0112dc7 contain a lot of changes, if you can review it once 0rb have confirmed that is solve most of the 'dependency hell' issue

File: external/os-guides/pentesting/kali.md
Comment (neowutran) [reply external/os-guides/pentesting/kali.md]: I have some difficulties with the English langage.. 
Reworded this to make it simpler

**Comment by 0rb677** [comment]
@neowutran did you add this ? `https://tools.kali.org/kali-metapackages` there is no `kali-linux-full` anymore.. you need install `kali-linux-default` then `kali-linux-large` and then 'kali-linux-everything`

**Comment by 0rb677** [comment]
@neowutran thanks for your work, now its works.
did you add this ? `https://tools.kali.org/kali-metapackages` there is no `kali-linux-full` anymore.. 
```
user@kali-rolling2:~$ cat /etc/apt/sources.list.d/qubes-r4.list
# Main qubes updates repository
deb [arch=amd64] https://deb.qubes-os.org/r4.0/vm bullseye main
#deb-src https://deb.qubes-os.org/r4.0/vm bullseye main

# Qubes updates candidates repository
deb [arch=amd64] https://deb.qubes-os.org/r4.0/vm bullseye-testing main
#deb-src https://deb.qubes-os.org/r4.0/vm bullseye-testing main

# Qubes security updates testing repository
deb [arch=amd64] https://deb.qubes-os.org/r4.0/vm bullseye-securitytesting main
#deb-src https://deb.qubes-os.org/r4.0/vm bullseye-securitytesting main

# Qubes experimental/unstable repository
#deb [arch=amd64] https://deb.qubes-os.org/r4.0/vm bullseye-unstable main
#deb-src https://deb.qubes-os.org/r4.0/vm bullseye-unstable main


# Qubes Tor updates repositories
# Main qubes updates repository
#deb [arch=amd64] tor+http://deb.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/r4.0/vm bullseye main
#deb-src tor+http://deb.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/r4.0/vm bullseye main

# Qubes updates candidates repository
#deb [arch=amd64] tor+http://deb.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/r4.0/vm bullseye-testing main
#deb-src tor+http://deb.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/r4.0/vm bullseye-testing main

# Qubes security updates testing repository
#deb [arch=amd64] tor+http://deb.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/r4.0/vm bullseye-securitytesting main
#deb-src tor+http://deb.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/r4.0/vm bullseye-securitytesting main

# Qubes experimental/unstable repository
#deb [arch=amd64] tor+http://deb.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/r4.0/vm bullseye-unstable main
#deb-src tor+http://deb.qubesosfasa4zl44o4tws22di6kepyzfeqv3tg4e3ztknltfxqrymdad.onion/r4.0/vm bullseye-unstable main
```
This is danger to remove
```
user@kali-rolling2:~$ sudo apt-get autoremove
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages will be REMOVED:
  debugedit enchant g++-8 geoip-database gir1.2-clutter-1.0
  gir1.2-clutter-gst-3.0 gir1.2-cogl-1.0 gir1.2-coglpango-1.0
  gir1.2-gtkclutter-1.0 gir1.2-gtksource-3.0 gir1.2-json-1.0 libaudio2
  libayatana-ido3-0.4-0 libbind9-161 libboost-system1.67.0 libclutter-1.0-0
  libclutter-1.0-common libclutter-gst-3.0-0 libclutter-gtk-1.0-0
  libcodec2-0.8.1 libcogl-common libcogl-pango20 libcogl-path20 libcogl20
  libcroco3 libcrystalhd3 libcupsimage2 libcwidget3v5 libdns1104 libdns1110
  libdouble-conversion1 libdw1 libenchant1c2a libevent-2.1-6 libexiv2-14
  libgeoip1 libgspell-1-1 libgtksourceview-3.0-1 libgtksourceview-3.0-common
  libgutenprint-common libgutenprint9 libhavege1 libigdgmm5 libilmbase23
  libisc1100 libisc1105 libisccc161 libisccfg163 libisl19 libjim0.77 libllvm7
  liblouis17 liblouisutdml8 liblwres161 libmng1 libmozjs-60-0 libmysofa0
  libnfs12 libnftables0 liboauth0 libopenexr23 libperl5.28
  libpulse-mainloop-glib0 libpython3.7 libqpdf21 libreadline7 librpm8
  librpmbuild8 librpmio8 librpmsign8 libruby2.5 libsnmp30 libsqlite0
  libstdc++-8-dev libtagc0 libusbmuxd4 libvpx5 libx265-165
  linux-headers-4.19.0-6-amd64 linux-headers-4.19.0-6-common
  linux-image-4.19.0-6-amd64 printer-driver-gutenprint python-daemon
  python-dbus python-iniparse python-libxml2 python-lockfile python-lzma
  python-pycurl python-qubesdb python-six python-sqlite python-sqlitecachec
  python-talloc python-urlgrabber python-xdg python3-croniter python3-crypto
  python3-dateutil python3-jinja2 python3-markupsafe python3-msgpack
  python3-psutil python3-tornado4 python3-tz python3-yaml python3-zmq
  qtchooser qtcore4-l10n qubes-core-agent-passwordless-root qubes-gpg-split
  qubes-img-converter qubes-input-proxy-sender qubes-mgmt-salt-vm-connector
  qubes-pdf-converter qubes-thunderbird qubes-usb-proxy rpm rpm-common
  rpm2cpio ruby-did-you-mean ruby2.5 salt-common salt-ssh
0 upgraded, 0 newly installed, 124 to remove and 0 not upgraded.
After this operation, 630 MB disk space will be freed.
Do you want to continue? [Y/n] 
user@kali-rolling2:~$ sudo apt-get autoremove
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages will be REMOVED:
  debugedit enchant g++-8 geoip-database gir1.2-clutter-1.0
  gir1.2-clutter-gst-3.0 gir1.2-cogl-1.0 gir1.2-coglpango-1.0
  gir1.2-gtkclutter-1.0 gir1.2-gtksource-3.0 gir1.2-json-1.0 libaudio2
  libayatana-ido3-0.4-0 libbind9-161 libboost-system1.67.0 libclutter-1.0-0
  libclutter-1.0-common libclutter-gst-3.0-0 libclutter-gtk-1.0-0
  libcodec2-0.8.1 libcogl-common libcogl-pango20 libcogl-path20 libcogl20
  libcroco3 libcrystalhd3 libcupsimage2 libcwidget3v5 libdns1104 libdns1110
  libdouble-conversion1 libdw1 libenchant1c2a libevent-2.1-6 libexiv2-14
  libgeoip1 libgspell-1-1 libgtksourceview-3.0-1 libgtksourceview-3.0-common
  libgutenprint-common libgutenprint9 libhavege1 libigdgmm5 libilmbase23
  libisc1100 libisc1105 libisccc161 libisccfg163 libisl19 libjim0.77 libllvm7
  liblouis17 liblouisutdml8 liblwres161 libmng1 libmozjs-60-0 libmysofa0
  libnfs12 libnftables0 liboauth0 libopenexr23 libperl5.28
  libpulse-mainloop-glib0 libpython3.7 libqpdf21 libreadline7 librpm8
  librpmbuild8 librpmio8 librpmsign8 libruby2.5 libsnmp30 libsqlite0
  libstdc++-8-dev libtagc0 libusbmuxd4 libvpx5 libx265-165
  linux-headers-4.19.0-6-amd64 linux-headers-4.19.0-6-common
  linux-image-4.19.0-6-amd64 printer-driver-gutenprint python-daemon
  python-dbus python-iniparse python-libxml2 python-lockfile python-lzma
  python-pycurl python-qubesdb python-six python-sqlite python-sqlitecachec
  python-talloc python-urlgrabber python-xdg python3-croniter python3-crypto
  python3-dateutil python3-jinja2 python3-markupsafe python3-msgpack
  python3-psutil python3-tornado4 python3-tz python3-yaml python3-zmq
  qtchooser qtcore4-l10n qubes-core-agent-passwordless-root qubes-gpg-split
  qubes-img-converter qubes-input-proxy-sender qubes-mgmt-salt-vm-connector
  qubes-pdf-converter qubes-thunderbird qubes-usb-proxy rpm rpm-common
  rpm2cpio ruby-did-you-mean ruby2.5 salt-common salt-ssh
0 upgraded, 0 newly installed, 124 to remove and 0 not upgraded.
After this operation, 630 MB disk space will be freed.
Do you want to continue? [Y/n] 
```
This is new metapackages
```
user@kali-rolling2:~$ sudo apt install kali-linux-
kali-linux-arm         kali-linux-default     kali-linux-large
kali-linux-core        kali-linux-everything  kali-linux-nethunter

```

**Comment by neowutran** [comment]
"Also maybe add prebuilt kali-linux-template to qubes repo?"
No, the goal is not to fix dependencies issues of all the [non supported] gnu/linux distributions. Maybe some modification could be done for the debian packaging to be less strict about the dependencies versions, but that is another subject 

For the packages that are selected for removal when you do a 'dist-upgrade', it is up to you to decide if you want to keep them or not.

It doesn't solve all the 'dependency hell', but I don't have any more 'quick win' idea

**Comment by unman** [comment]
Give me a chance to look through this and finalise- most likely in the morning, Looks good, I think.

**Comment by 0rb677** [comment]
@neowutran new issue
```
The following packages have unmet dependencies:
 libc6-dev : Breaks: libgcc-8-dev (< 8.4.0-2~) but 8.3.0-6 is to be installed
E: Error, pkgProblemResolver::Resolve generated breaks, this may be caused by held packages.

```

**Comment by neowutran** [comment]
Some issues [improvement to be done / ... ] with this pull request ?

**Comment by andrewdavidwong** [comment]
> Some issues [improvement to be done / ... ] with this pull request ?

Looks like we're waiting on @unman's (re-)review.

**Comment by minic0m** [comment]
@neowutran also need to remove gcc-8 on a template before update to kali-rolling
```
sudo apt-get remove libgcc-8-dev && sudo apt-get install libc6-dev
```

File: external/os-guides/pentesting/kali.md
Comment (andrewdavidwong) [inline external/os-guides/pentesting/kali.md:131]: This is a bit cryptic. Do you mean something like, "This step is required until X is fixed/implemented"?

Comment (neowutran) [reply external/os-guides/pentesting/kali.md:131]: Something like that yes, with more words: 
This kind of issues will popup and disappear without notice.
It depends on the differences of dependencies between the kali repository, the testing qubes repository and the dependencies issues related to the upgrade from stable debian to testing debian/kali.

In comparison, the others steps are required regardless of the dependencies version of kali & qubes repository.

IDK how to make that easily understandable

Comment (andrewdavidwong) [reply external/os-guides/pentesting/kali.md:131]: Ok, thanks for the explanation. I will provide an edit that attempts to make it clearer.

Comment (andrewdavidwong) [reply external/os-guides/pentesting/kali.md:131]: Actually, I'm not sure whether "this step" refers to step 8 or step 9.

Comment (andrewdavidwong) [reply external/os-guides/pentesting/kali.md:131]: I also don't know what you're referring to what you say "this kind of issue."

I suggest that you add the type of more verbose explanation that you provided here.

Comment (neowutran) [reply external/os-guides/pentesting/kali.md]: Re-worded this part to be more specific

**Comment by neowutran** [comment]
@unman when you have some time to review that ^^

**Comment by unman** [comment]
On Mon, Aug 31, 2020 at 03:19:20AM -0700, Yukikoo wrote:
> @unman when you have some time to review that ^^
> 

Apologies all round - have had little time for any of this.

**Comment by andrewdavidwong** [comment]
Would it be appropriate to merge now, and further improvements can be made later? Or do you all think it would be better to wait until @unman has time for another review?

**Comment by unman** [comment]
On Sun, Sep 13, 2020 at 05:44:31AM -0700, Andrew David Wong wrote:
> Would it be appropriate to merge now, and further improvements can be made later? Or do you all think it would be better to wait until @unman has time for another review?
> 

Too late now - I'll d it first thing in the morning.

File: external/os-guides/pentesting/kali.md
Comment (unman) [inline external/os-guides/pentesting/kali.md:24]: Make this a link to HVM creation page

File: external/os-guides/pentesting/kali.md
Comment (unman) [inline external/os-guides/pentesting/kali.md:42]: You havent said that you need to update the template from "stable".
Without that this wont work

File: external/os-guides/pentesting/kali.md
Comment (unman) [inline external/os-guides/pentesting/kali.md:87]: Add warning about any changes in Debian repository naming conventions.
For example in change to bullseye you have to take account of the change to bullseye-security which your `sed` example wont cover

File: external/os-guides/pentesting/kali.md
Comment (unman) [inline external/os-guides/pentesting/kali.md:131]: dependency issue

File: external/os-guides/pentesting/kali.md
Comment (unman) [inline external/os-guides/pentesting/kali.md:132]: What about:
Such issues arise because of the differences of dependencies in packages from the Kali repository, the Qubes testing repository and the Debian testing repository.

File: external/os-guides/pentesting/kali.md
Comment (unman) [inline external/os-guides/pentesting/kali.md:147]: I *would* at least point to the available metapackages here.

**Review by unman — CHANGES_REQUESTED** [review]

File: external/os-guides/pentesting/kali.md
Comment (neowutran) [reply external/os-guides/pentesting/kali.md:150]: Ok, added a link to the official doc

File: external/os-guides/pentesting/kali.md
Comment (neowutran) [reply external/os-guides/pentesting/kali.md]: Thanks, replaced the content

File: external/os-guides/pentesting/kali.md
Comment (neowutran) [reply external/os-guides/pentesting/kali.md:88]: I completely remove every link to any Debian official repository and replace it with Kali repository, so no issue with naming convention. The only remaining thing related to `buster` / `bullseye` is Qubes OS repository

File: external/os-guides/pentesting/kali.md
Comment (neowutran) [reply external/os-guides/pentesting/kali.md]: ok, done

**Comment by andrewdavidwong** [comment]
I'm about to move external documents like this one out of `qubes-doc` (see https://github.com/QubesOS/qubes-issues/issues/4693). However, I don't want to move this out from under you just as you're finalizing the PR. It would probably be easier if we merged this first.

**Comment by andrewdavidwong** [comment]
Just checking in to try to understand the status of this PR. Is it currently undergoing further work from @neowutran or further review from @unman? Or perhaps both?

**Comment by neowutran** [comment]
More like waiting for a comment on this https://github.com/QubesOS/qubes-doc/pull/989#discussion_r492725504 
I see that we are slowing you down with the doc splitting. 
I don't mind if you close this pull request, I would open a new in the new "community doc" repo and would add a link to this closed pull request for references

**Comment by andrewdavidwong** [comment]
No, I wasn't trying to rush you. I just wanted to avoid a situation where each person is waiting on the other due to lack of communication. :slightly_smiling_face: 

If work is still ongoing, please go ahead, and don't feel the need to rush.

**Comment by andrewdavidwong** [comment]
How are things progressing on this? :slightly_smiling_face:

**Comment by dud1337** [comment]
I've followed the new guide from this pull request on creating a kali-rolling instance from debian-10 and it worked for me. That gcc-8-base issue was an annoyance and I appreciate the temporary solution detail being included here.

**Comment by andrewdavidwong** [comment]
> I've followed the new guide from this pull request on creating a kali-rolling instance from debian-10 and it worked for me. That gcc-8-base issue was an annoyance and I appreciate the temporary solution detail being included here.

Thanks for testing it! In that case, I'll proceed to merge this. Since it's external documentation, we're mainly just looking for confirmation that it works.

**Pull request merged**
